packages/hadron-build/commands/upload.js

```diff
@@ -1,139 +1,466 @@
+const { promises: fs } = require('fs');
+const { deepStrictEqual } = require('assert');
 const { Octokit } = require('@octokit/rest');
 const { GithubRepo } = require('@mongodb-js/devtools-github-repo');
+const { diffString } = require('json-diff');
+const download = require('download');
+const Target = require('../lib/target');
+const {
+  getKeyPrefix,
+  downloadManifest,
+  uploadAsset,
+  uploadManifest
+} = require('../lib/download-center');
 
 const cli = require('mongodb-js-cli')('hadron-build:upload');
 const abortIfError = cli.abortIfError.bind(cli);
-const Target = require('../lib/target');
+const root = path.resolve(__dirname, '..', '..', '..');
 
-const downloadCenter = require('../lib/download-center');
+async function checkAssetsExist(paths) {
+  await Promise.all(
+    paths.map((assetPath) => {
+      return fs.stat(assetPath);
+    })
+  );
+  return true;
+}
 
-async function maybePublishGitHubRelease(target) {
-  if (!process.env.GITHUB_TOKEN) {
-    cli.warn('Skipping publish release because process.env.GITHUB_TOKEN not set.');
-    return;
+function isBeta(id) {
+  return /-beta.\d+/.test(id);
+}
+
+function isStable(id) {
+  return !isBeta(id);
+}
+
+const distributionLabel = {
+  'compass-readonly': 'Readonly Edition',
+  'compass-isolated': 'Isolated Edition'
+};
+
+const channelLabel = {
+  stable: 'Stable',
+  beta: 'Beta'
+};
+
+function versionId(version, distribution) {
+  return [version, distribution.replace(/compass\-?/, '')]
+    .filter(Boolean)
+    .join('-');
+}
+
+function readableVersionName(version, channel, distribution) {
+  const desc = [distributionLabel[distribution], channelLabel[channel]]
+    .filter(Boolean)
+    .join(' ');
+  return `${version} ${desc ? `(${desc})` : ''}`.trim();
+}
+
+function readablePlatformName(arch, platform, fileName) {
+  let name = null;
+
+  switch (`${platform}-${arch}`) {
+    case 'darwin-x64':
+      name = 'macOS 64-bit (10.10+)';
+      break;
+    case 'darwin-arm64':
+      name = 'macOS arm64 (M1) (11.0+)';
+      break;
+    case 'win32-x64':
+      name = 'Windows 64-bit (7+)';
+      break;
+    case 'linux-x64':
+      name = fileName.endsWith('.rpm')
+        ? 'RedHat 64-bit (7+)'
+        : 'Ubuntu 64-bit (14.04+)';
+      break;
+    default:
+      throw new Error(
+        `Unexpected asset for the download center: ${platform} ${arch} ${fileName}`
+      );
+  }
+
+  if (fileName.endsWith('.zip')) {
+    name += ' (Zip)';
+  }
+
+  if (fileName.endsWith('.msi')) {
+    name += ' (MSI)';
+  }
+
+  return name;
+}
+
+function generateVersionsForAssets(assets, version, channel) {
+  return Target.supportedDistributions.map((distribution) => {
+    return {
+      _id: versionId(version, distribution),
+      version: readableVersionName(version, channel, distribution),
+      platform: assets
+        .filter((asset) => {
+          return asset.config.distribution === distribution;
+        })
+        // eslint-disable-next-line no-shadow
```

Comment:
hadron-build is on the "old" eslint config 😞 I didn't want to make the diff more convoluted, but will do a follow up and update it to the new one (it also will allow us to eventually move the code here to typescript) and will clean up unused eslint rules

Reply:
Ah no worries – if we’re already using a newer eslint config elsewhere, that also sounds good. I just felt like I’d run into this rule occasionally in the past and assumed that it would affect Compass more broadly, that’s all